### ISSUE TITLE
Syntax error programs throw unhandled exceptions

### ISSUE THREAD

**Issue by dabbler0** [droplet-editor/droplet#21]
The reparse operation that happens every time we drop a block throws an unhandled error if the resulting combination is a syntax error. Example:
```
1 + 1;
function hello(a) {
  return a;
}
```
Drag the `1+1` block to make:
```
function hello(1 + 1) {
  return a;
}
```
In this case the drop appears not to "complete"; the `1+1` block will continue to be dragged until another mouse event. This is because the process that cleans up the end of the drag has thrown an error while trying to reparse the function block and had not recovered.

Ideally, in this situation, Droplet would undo the drop or forbid the drop to begin with.

**Comment by sehgalvibhor** [comment]
Hi,

I want to work on this issue. Has it been already fixed or not? Whenever i try the program example you mentioned above, the (1+1) block up ends like this
function(a){
      return '1+1';
}

The block automatically goes in the return statement of the loop. Does that need to be fixed?
Thanks.

**Comment by dabbler0** [comment]
Hi @sehgalvibhor,

Yep, this issue has was fixed in PR #64. We'd still love to have you work on an issue, though, if you like -- #93 (an Internet Explorer compatability issue) is still open, as well as #56 (a rendering issue)!
